### PR TITLE
Fixes 378: option added to edit the sub project

### LINKED ISSUE #378: Allow users to update sub projects registered to a master project
### This is a
- [ ] Bug report
- [x] Feature request

### Current behaviour

User can add sub project only while registering sub project.
### Expected behavior

user should have the option to add or remove sub project after registering the project.

### Steps to reproduce
1.
2.
3.

### PR DESCRIPTION
**Pull request by sch00lb0y**
## Description

I have added option to edit the sub project 
## Related Issue

close #378 
## How Has This Been Tested?

http://demo08.herokuapp.com/
## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by pri22296** [comment]
@sch00lb0y Registration failed while adding subrepo.

**Comment by sch00lb0y** [comment]
@pri22296 I just tested. it is working. I have cleared the db. remove the webhook and please check once again

**Comment by pri22296** [comment]
@sch00lb0y Worked. One suggestion though. Don't redirect to dashboard every time I add or remove a repo.

**Comment by imujjwal96** [comment]
Change the title to:- `Add a subproject`
image

**Comment by imujjwal96** [comment]
as @pri22296 pointed, do not redirect to the dashboard! Stay in the settings page after the operation

File: public/scripts/search.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/**
+ * Search for repositories
+ */
+var subProjectId = 0; // sub projects dynamic id
+var repositorySelectIds = []; // array for storing repositories select element
+
+var registerDisableChecker = function () {
+  var flag = true; // flag to check whether any of the select is
```
Comment (imujjwal96) [inline public/scripts/search.js]: any of the select is....?

File: public/scripts/search.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+/**
+ * Search for repositories
+ */
+var subProjectId = 0; // sub projects dynamic id
+var repositorySelectIds = []; // array for storing repositories select element
+
+var registerDisableChecker = function () {
```
Comment (imujjwal96) [inline public/scripts/search.js]: add documentation of this function

File: public/scripts/search.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  })
+  if (flag) {
+    styles.enableButton("btnRegister");
+  } else {
+    styles.disableButton("btnRegister");
+  }
+}
+
+var search = function (searchBarInput, searchResultDiv, tempSubProjectId) {
```
Comment (imujjwal96) [inline public/scripts/search.js]: add documentation for tthis function

File: routes/index.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ var router = express.Router();
 var authMiddleware = require("../middleware/auth");
 
 var hostname = process.env.HOSTNAME || 'yaydoc.herokuapp.com';
+var github = require("../backend/github")
```
Comment (imujjwal96) [inline routes/index.js]: add semicolon :)

**Comment by imujjwal96** [comment]
Also, change the commit message from `Fixes 378.......` to `Fixes #378` Most of your PR's doesn't close the issues automatically.

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96  updated.

**Pull request merged**